### ISSUE TITLE
How to migrate "data[..."  to v4?

### ISSUE THREAD

**Issue by IdanAdar** [mikefarah/yq#1765]
What is the v4 equivalent for this line?

```
yq r cluster-info.yaml 'data[cluster-config.json]' | yq r - cluster_id)
```

Is it simply
```
yq  'data[cluster-config.json]' cluster-info.yaml | yq 'cluster_id')
```

Another question:
If I have no $variable, I can use single quotes, but if I do have a variable I should use double quotes to ensure expansion in shell?

**Comment by mikefarah** [comment]
I'm don't quite understand what that is doing - do you have a sample data and the expected output? 

I recommend always using single quotes - if you have any variables use the built in `strenv` and `env` operators - otherwise it can get pretty complex with quote escaping. See https://mikefarah.gitbook.io/yq/operators/env-variable-operators for details.

**Comment by IdanAdar** [comment]
Here's an example from past ticket of mine: https://github.com/mikefarah/yq/issues/181

**Comment by mikefarah** [comment]
Ah I see, also nice to hear from you again Idan :)

yq v4 has quite different syntax from v3 - instead of taking a command and path it takes an expression string. This does make it a little harder to learn (unless you have familiarity with jq) but you can do some much more powerful things with it.  

TLDR:

```
yq '.data["cluster-config.json"] | from_json | .cluster_id' file.yaml
```

- `.data["cluster-config.json"] ` will read the entry in the yaml file - which itself is a string encoded json
- `from_json` will parse that string
- `.cluster_id` will read the nested property.
